lec04-dplyr.Rmd

```diff
@@ -1,4 +1,724 @@
+    tally() %>% 
+    arrange(desc(n)) %>% 
+    head(5)
+```
+
+Be careful not to include anything that would split the group into subgroups,
+such as `sex`, `year` etc.
+
+> #### Challenge
+>
+> 1. How many individuals were caught in each `plot_type` surveyed?
+>
+> 2. You saw above how to count the number of individuals of each `sex` using a
+> combination of `group_by()` and `tally()`. How could you get the same result
+> using `group_by()` and `summarize()`? Hint: see `?n`.
+>
+
+<!---
+```{r, echo=FALSE}
+## Answer 1
+surveys %>%
+    group_by(plot_type) %>%
+    tally
+
+## Answer 2
+surveys %>%
+  group_by(sex) %>%
+  summarize(n = n())
+  
+```
+--->
+
+
+## Plotting with ggplot2
+
+**`ggplot2`** is a plotting package that makes it simple to create complex plots
+from data frames. The name **`ggplot2`** comes from its inspiration, the book "A
+grammar of graphics", and its main goal is to that allow programmer to express
+their desired outcome on a high level instead of telling the computer every
+detail about what will happen. For example, you would say "color my data by
+species" instead of "go through this data frame and plot any observations of
+species1 in blue, any observations of species2 in red, etc". Thanks to this
+functional way of interfaces with data, only minimal changes are required if the
+underlying data change or to change the type of plot. This helps in thinking
+about the data and creating publication quality plots with minimal amounts of
+adjustments and tweaking.
+
+ggplot graphics are built step by step by adding new elements, or layers. Adding layers in
+this fashion allows for extensive flexibility and customization of plots. To
+build a ggplot, we need to:
+
+1. Use the `ggplot()` function and bind the plot to a specific data frame using the  
+      `data` argument
+
+```{r}
+ggplot(data = surveys)
+```
+
+Rememer that the argument names of a function can be left out, if the arguments are provided in the right order.
+
+```{r}
+ggplot(surveys)
+```
+
+2. Define aesthetics (`aes`), by selecting the variables to be plotted and the  
+      variables to define the presentation such as plotting size, shape color, etc.
+
+```{r}
+ggplot(surveys, aes(x = weight, y = hindfoot_length))
+```
+
+3. Add `geoms` -- geometrical objects as a graphical representation of the data 
+in the plot (points, lines, bars). **`ggplot2`** offers many different geoms; we
+will use a few common ones today, including:
+      * `geom_point()` for scatter plots, dot plots, etc.
+      * `geom_line()` for trend lines, time-series, etc.  
+      * `geom_histogram()` for histograms
+
+To add a geom to the plot use `+` operator. Because we have two continuous variables,  
+let's use `geom_point()` first:
+
+```{r first-ggplot}
+# If this takes way too long on your machine, create a subset from a random
+# sample of a suitable size and continue working with this instead of `survey`.
+#survey_subset <- sample_n(surveys, size = 5000)
```

Comment:
might be a good idea to do for the website generation, to reduce the Travis CI build time.

Reply:
Yes, that makes sense. I didn't want people to end up with different results... I guess I could set the seed for sample. But then we will not run into the overplotting problem so that we can fix it. Or maybe we would still. I will check with different sample sizes after adjusting for the rest of your feedback.

Reply:
It's not a huge deal. The build time is still <3 min so really, this might be premature optimization. Don't worry about it :P